net/SslSocket.hpp

```diff
@@ -357,26 +362,37 @@ class SslStreamSocket final : public StreamSocket
                 errno = last_errno; // Restore errno.
                 return rc;
 
-            case SSL_ERROR_WANT_CONNECT:
+            // Retry.
+            case SSL_ERROR_WANT_CONNECT: // 7
                 LOG_TRC("SSL error (" << context << "): WANT_CONNECT (" << sslError
                                       << "): " << bioErrStr);
                 errno = last_errno; // Restore errno.
                 return rc;
 
-            case SSL_ERROR_WANT_ACCEPT:
+            // Retry.
+            case SSL_ERROR_WANT_ACCEPT: // 8
                 LOG_TRC("SSL error (" << context << "): WANT_ACCEPT (" << sslError
                                       << "): " << bioErrStr);
                 errno = last_errno; // Restore errno.
                 return rc;
 
-            case SSL_ERROR_WANT_X509_LOOKUP:
-                // Unexpected.
+            // Unexpected: happens only with SSL_CTX_set_client_cert_cb().
+            case SSL_ERROR_WANT_X509_LOOKUP: // 4
                 LOG_TRC("SSL error (" << context << "): WANT_X509_LOOKUP (" << sslError
                                       << "): " << bioErrStr);
                 errno = last_errno; // Restore errno.
                 return rc;
 
-            case SSL_ERROR_SYSCALL:
+            // Unexpected: happens only with SSL_CTX_set_client_cert_cb().
+            case SSL_ERROR_WANT_CLIENT_HELLO_CB: // 11
+                LOG_TRC("SSL error (" << context << "): WANT_X509_LOOKUP (" << sslError
+                                      << "): " << bioErrStr);
+                errno = last_errno; // Restore errno.
```

Comment:
Was this the functional change / fix ? =)

Reply:
It's here for completeness.